### ISSUE TITLE
Portable App GitHub Repo Template

### ISSUE THREAD

**Issue by Adhjie** [Numstr/Obsidian-Portable#3]
So anyone could get the latest release without waiting the maintainer of portable apps in PortableApps.com

1. So first question, could you share the template on how to make any app portable like this repo? maybe a repo that is a tutorial on how to make any apps portable called "Portable App Template".

3. Second question, how does this one always fetch the latest release just by using updater.bat file?

4. Is this inspired from PortableApps platform but differs or is the exact same mechanism?

5. How about ensuring it, being safe e.g., the .apk match the signing certificate hash by using AppVerifier/apksigner in desktop?

6a. ~~(this is a bit sidetrack but it is possible to download a lot of apps straight from GitHub through an app called Obtainium and verify it using AppVerifier app that is integrated with Obtainium, now I never heard mobile portable app but this indeed is interesting, additional bonus question?~~
~~This is entirely not on topic since it's mobile, but I guess putting this in package manager is a way to get this always updated, or an auto-update feature is on the roadmap)~~

Actual question, could AppVerifier be integrated with this app?
(Obtainium desktop port is a great idea to rival package manager but I think that's too advance for now.)

6b. Additional question about updater, is the update manual, is there an auto-updater, or is it easier to put this in package manager with a volunteer maintainer?

6c. And most package managers have checks like Winget, Chocolatey, and also PortableApps with Virustotal scan, how to check the safeness of this app?

**Comment by Adhjie** [comment]
Oh, yeah. I have just realized. since this is more of a suggestion. Isn't GitHub discussion feature more appropriate to be opened for this type of question, rather than issue tab only?

**Comment by Numstr** [comment]
Im make this for my self to fast check latest version and download it. 90% of time you dont need to update launcher `AppNameLauncher.exe`, just update app located in `App\AppName`. Also you can download latest release from github manually, unpack it used 7zip and replace all files in `App\AppName`.

> So first question, could you share the template on how to make any app portable like this repo? maybe a repo that is a tutorial on how to make any apps portable called "Portable App Template".
> Is this inspired from PortableApps platform but differs or is the exact same mechanism?

It use PortableApps.com Launcher to generate launcher, each app must be described (each app leave different trash) in `App/AppInfo/Launcher/AppNamePortable.ini` [here](https://portableapps.com/manuals/PortableApps.comLauncher/) info how to do this, also you can find tutorial on youtube or google.
> Second question, how does this one always fetch the latest release just by using updater.bat file?

`Updater.bat` is just script that use some linux utils located in `App/Utils` to check for latest version from obsidian github repo, download it and unpack. If devs change package name or structure `Updater.bat` can be broken. Also, not all installers can be unpack easily.

>How about ensuring it, being safe e.g., the .apk match the signing certificate hash by using AppVerifier/apksigner in desktop?

You can check hash/certificate for `Obsidian.exe` located in App/Obsidian. 

Creating something like Obtainium takes a lot of time, which I dont have.

**Comment by Adhjie** [comment]
Okay for now I will check the link and do this myself manually. Since I don't have the time either to make a portable launcher like this repository.

**Comment by Adhjie** [comment]
@Numstr 
Okay, so to summarize. I just need to change the info in
```
App/AppInfo/Launcher/AppNamePortable.ini
```
and
```
Updater.bat
```
to make any portableapps' app self-updatable by myself?
or Do I need to mimic every settings that exist for each particular app?
e.g., DocFetcher and Chrome have specific settings that needs to be set to true for some thing to work. Though, Chrome is probably not ever going to be fixed, because of Chromium encryption limitation.

**Comment by Adhjie** [comment]
Also where can I read more on the list of commandline parameters for ObsidianPortable?
I see it in ObsidianPortable.ini in the ObsidianPortable\ directory.

**Comment by Numstr** [comment]
> to make ~~any~~ portableapps' app self-updatable by myself?

Not any app. To update yourself, you need to change the BAT file, but you need to know what exactly needs to be changed. It will be useful to know a little CMD.

> Chrome is probably not ever going to be fixed, because of Chromium encryption limitation.

In the latest builds of Chromium, they fixed the cookie encryption (add --disable-encryption to the arguments). Ungoogled Chromium from my repo is now fully portable (need fresh profile). Dont know about Google Chrome and other chromium based browsers.

> this is the only portableapps obsidian portable that self-update

Obsidain has a built-in self-updating mechanism, so no.

> why not post this as official app in portableApps? If you're busy, could I fork this and post this to PortableApps?

Do as you wish, the repo is open. But this makes no sense, you can find packages of obsidian dev-test on the forum. The difference is small except for the presence of a BAT file. But if you talk about including it in the official portable app platform, they don't accept it because there are extra utils and BAT file present.

> Also where can I read more on the list of commandline parameters (I mean) AdditionalParameters for ObsidianPortable? (Tho, I want to know that CLI parameters too)

This parametrs are application specific. Obsidian use Electron so check parametrs for electron apps. Some work, some don't.

I don't really like answering questions and making guides. If you want a specific app, tell me what app is, I check it and maybe make a repo like that.

**Comment by Adhjie** [comment]
> > to make ~any~ portableapps' app self-updatable by myself?
> 
> Not any app. To update yourself, you need to change the BAT file, but you need to know what exactly needs to be changed. It will be useful to know a little CMD.

Okay, I'll search all the  commands in the Update.bat file, and learn it.

> > Chrome is probably not ever going to be fixed, because of Chromium encryption limitation.
> 
> In the latest builds of Chromium, they fixed the cookie encryption (add --disable-encryption to the arguments). Ungoogled Chromium from my repo is now fully portable (need fresh profile). Dont know about Google Chrome and other chromium based browsers.

does it contain all the history, no history will be gone? how about password? and does it support sync with chrome in mobile. or it only supports sync with mobile ungoogled chromium fork? 

> > this is the only portableapps obsidian portable that self-update
> 
> Obsidain has a built-in self-updating mechanism, so no.

I thought, it's a dangerous thing to do that with portableApps (from reading Firefox support page in PA website), is it safe to do it with portable obsidian from PortableApps template?

> > why not post this as official app in portableApps? If you're busy, could I fork this and post this to PortableApps?
> 
> Do as you wish, the repo is open. But this makes no sense, you can find packages of obsidian dev-test on the forum. The difference is small except for the presence of a BAT file. But if you talk about including it in the official portable app platform, they don't accept it because there are extra utils and BAT file present.

Makes sense, the updating part is very different from PA version that needs maintainer, tho PR to the maintainer's GitHub would solved it; and if they go offline. A fork is doable, IG.

> > Also where can I read more on the list of commandline parameters (I mean) AdditionalParameters for ObsidianPortable? (Tho, I want to know that CLI parameters too)
> 
> This parametrs are application specific. Obsidian use Electron so check parametrs for electron apps. Some work, some don't.

Okay, thanks for the answer. I'll look into electron parameters.

> I don't really like answering questions and making guides. If you want a specific app, tell me what app is, I check it and maybe make a repo like that.

Okay, This is awesome. I'll make a list later. If it's not possible don't push it for now. But if it's possible thanks, self-updating portable apps are awesome. great addition besides winget, chocolatey, scoop (mostly portable but i cant get it to work in my machine), msys2 for installed apps.

I'm gonna list some on my devices, just cross the items if it's actively maintained in PA website (I havent checked them all yet). this is gonna be a drafts (wait, on second thought. I'll open new issue for this. since this issue question is answered).

Again thank you so much. This portable apps really save my cellular data, all the self-updating installed apps are not mobile data friendly (e.g., most installed browsers are self-updating).

**Comment by Adhjie** [comment]
Alright, I'm gonna limit the scope. I think my most imperative app to update is Firefox Browser.
So is it generally a good idea to download and extract firefox to the firefox portable directory myself, or is that dangerous. and only download and install from portableApps?

I see chromeplus project using chrome offline installer, is it possible to manually install firefox 133.0 to the portable firefox directory or should I be patient and wait for portableApps to reach FF 133.0

tl;dr
how to make an updater.bat, self-update using auto-extraction script for firefox portable?
My most critical apps are firefox, syncthing KeePassXC, fallback to KeePass, and Obsidian.
but honestly only firefox is the most critical right now. since browser is updated fairly frequent.

**Comment by Numstr** [comment]
> Alright, I'm gonna limit the scope. I think my most imperative app to update is Firefox Browser.
So is it generally a good idea to download and extract firefox to the firefox portable directory myself, or is that dangerous. and only download and install from portableApps?

Its safe, unpack it to App\Firefox or App\Firefox64 for x64.

Found my old script for Firefox, it still works https://gist.github.com/Numstr/fc1b46beea7c845cbde764198f947def also need busybox and 7za in App\Utils. If you use Firefox from portable apps, remove App\Firefox64 folder.

**Comment by Adhjie** [comment]
Alright, I'll close this. Because, asking params without reading docs felt too demanding for a private repos.
thanks for your fork and tweak, to make a truly self-portable apps by adding busybox and 7z to the PortableApps format. Great fork!
